scripts/west_commands/runners/openocd.py

```diff
@@ -422,12 +427,27 @@ def do_attach_debug_rtt(self, command, **kwargs):
         elif command == 'rtt':
             self.print_rttserver_message()
             server_proc = self.popen_ignore_int(server_cmd)
+
             try:
                 # run the binary with gdb, set up the rtt server (runs to completion)
                 subprocess.run(gdb_cmd)
+
+                if os_name != 'nt':
+                    # Save the terminal settings
+                    fd = sys.stdin.fileno()
+                    new_term = termios.tcgetattr(fd)
+                    old_term = termios.tcgetattr(fd)
```

Comment:
1. This needs to be moved to before the `try:`  block
2. There needs to be an `else:` added to set these variables to `None`

Otherwise consider what would happen if subprocess.run() failed: we'd get an unbound local variable error in the `finally`.

How was this tested?